pkg/sql/crdb_internal.go

```diff
@@ -9052,25 +9058,53 @@ CREATE TABLE crdb_internal.cluster_replication_node_streams (
 			return tree.NewDInterval(duration.Age(now, t), types.DefaultIntervalTypeMetadata)
 		}
 
+		shortenLogical := func(d *tree.DDecimal) *tree.DDecimal {
+			var tmp apd.Decimal
+			d.Modf(nil, &tmp)
+			if tmp.IsZero() {
+				if _, err := tree.DecimalCtx.Quantize(&tmp, &d.Decimal, -1); err == nil {
+					d.Decimal = tmp
+				}
+			}
+			return d
+		}
```

Comment:
Just to confirm: It looks like this will trim the fractional component if it is zero (which is the very common case). So it transforms something like 1718017491135909000.0000000000 to 1718017491135909000.0?

Reply:
Exactly.